src/hd.rs

```diff
@@ -526,10 +489,8 @@ impl Serialize for HDWallet {
         S: serde::Serializer,
     {
         let mut rgb = serializer.serialize_struct("HDWallet", 3)?;
-        rgb.serialize_field("xpub", &self.master)?;
-        rgb.serialize_field("origin", &KeyOrigin::from_extkey(&self.master))?;
+        rgb.serialize_field("descriptor", &self.descriptor.0)?;
         rgb.serialize_field("network", &self.network)?;
-        rgb.serialize_field("script_type", &self.script_type)?;
```

Comment:
We technically don't need the custom serializer for `HDWallet` now that it doesn't have the virtual `origin` field, this could simply `#[derive(Serialize)]`.

But since this will eventually be added back for compatibility with xpub-based usage, maybe better to keep this here for now anyway.